python/vimsence.py

```diff
@@ -261,3 +265,49 @@ def get_directory():
     '''
 
     return re.split(r'[\\/]', vim.eval('getcwd()'))[-1]
+
+def get_dir_path():
+    '''Get absolute path of current dir in form of list
+    :return: list of directory in order
+    eg: INPUT : "/home/anurag/.vim/bundle/vimsence/python"
+        OUTOUT: ['home', 'anurag', '.vim', 'bundle', 'vimsence', 'python']
+    '''
+    return re.split(r'[\\/]', vim.eval('getcwd()'))[1:]
+    
+def get_git_info():
```

Comment:
Is there any information about how other (similar) projects get the URL's for Git repo's? It seems like this is quite error prone in this way and I'm interested to see how other projects tackle that.

Reply:
Sure I'll check how the VS Code extension and neovim extension do it.

Reply:
Yeah I just went through the source code of COC-DISCORD-RCP for neovim and they do use a different approach. They use git command to get the address. I am gonna change the approach.

Reply:
Finally, it's done. I changed the entire approach to using the git commands. No looping tough folders, no searching .git dir no reading the config file.
**All the subprocess git stuff is done in this commit 4c4c112**
I didn't use the regex which they have used cuz I didn't understand that. They used the following regex.
```ts
const matched =
      /^(?:git@|https?:\/\/)(?<provider>[a-zA-Z_.~-]+\.[a-z]{2,})(?::|\/)(?<user>.*?)\/(?<repo>.*)$/.exec(
      remoteUrl.stdout.trim()
      );
```
Seems like just one regex statement does both HTTPS and SSH parsing. Tho my regex is also very strong and works.